features/include/pcl/features/impl/moment_of_inertia_estimation.hpp

```diff
@@ -606,37 +606,37 @@ pcl::MomentOfInertiaEstimation<PointT>::setIndices (const PointIndicesConstPtr&
 template <typename PointT> void
 pcl::MomentOfInertiaEstimation<PointT>::setIndices (std::size_t row_start, std::size_t col_start, std::size_t nb_rows, std::size_t nb_cols)
 {
-  if ((nb_rows > input_->height) || (row_start > input_->height))
+  if ((nb_rows > static_cast<uindex_t>(input_->height)) || (row_start > static_cast<uindex_t>(input_->height)))
```

Comment:
I'd suggest adding a note to overhaul the API of such functions via the following discussed process:
1. Add API for `index_t`
2. Deprecate + SFINAE out existing function
3. Later remove deprecation

Reply:
Should this note appear in one central location (in `point_cloud.h`) or under each affected function?

Reply:
Either an itemized list in a central location, or in each affected location. Itemization or individual note will make it easier to find those functions

Reply:
Here's a full list of functions where we're currently casting to `uindex_t`. I'm not sure if I should spam `point_cloud.h` with a note this large :flushed:
```cpp
/*
API of the following functions (where index_t variables are compared with unsigned variables) must be modified;

common/include/pcl/common/impl/io.hpp -  
copyPointCloud (const pcl::PointCloud<PointT> &cloud_in, pcl::PointCloud<PointT> &cloud_out,
                int top, int bottom, int left, int right, pcl::InterpolationType border_type, const PointT& value)

common/include/pcl/common/impl/spring.hpp -
duplicateColumns (const PointCloud<PointT>& input, PointCloud<PointT>& output,
                  const std::size_t& amount)
duplicateRows (const PointCloud<PointT>& input, PointCloud<PointT>& output,
               const std::size_t& amount)
mirrorColumns (const PointCloud<PointT>& input, PointCloud<PointT>& output,
               const std::size_t& amount)
mirrorRows (const PointCloud<PointT>& input, PointCloud<PointT>& output,
            const std::size_t& amount)
mirrorRows (const PointCloud<PointT>& input, PointCloud<PointT>& output,
            const std::size_t& amount)
deleteCols (const PointCloud<PointT>& input, PointCloud<PointT>& output,
            const std::size_t& amount)

common/include/pcl/impl/pcl_base.hpp -
pcl::PCLBase<PointT>::setIndices (std::size_t row_start, std::size_t col_start, std::size_t nb_rows, std::size_t nb_cols)

common/src/gaussian.cpp - 
pcl::GaussianKernel::convolveRows (const pcl::PointCloud<float>& input,
                                   const Eigen::VectorXf& kernel,
                                   pcl::PointCloud<float>& output)
pcl::GaussianKernel::convolveCols (const pcl::PointCloud<float>& input,
                                   const Eigen::VectorXf& kernel,
                                   pcl::PointCloud<float>& output)

features/include/pcl/features/impl/integral_image_normal.hpp - 
pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::initSimple3DGradientMethod ()
pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::initCovarianceMatrixMethod ()
pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::initAverage3DGradientMethod ()
pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::computePointNormal (
    const int pos_x, const int pos_y, const unsigned point_index, PointOutT &normal)

features/include/pcl/features/impl/moment_of_inertia_estimation.hpp
pcl::MomentOfInertiaEstimation<PointT>::setIndices (std::size_t row_start, std::size_t col_start, std::size_t nb_rows, std::size_t nb_cols)

features/src/range_image_border_extractor.cpp - 
RangeImageBorderExtractor::extractLocalSurfaceStructure ()

filters/impl/fast_bilateral_omp.hpp - 
pcl::FastBilateralFilterOMP<PointT>::applyFilter (PointCloud &output)

io/include/pcl/io/impl/pcd_io.hpp - 
pcl::PCDWriter::writeASCII (const std::string &file_name, const pcl::PointCloud<PointT> &cloud, 
                            const int precision)
pcl::PCDWriter::writeASCII (const std::string &file_name, 
                            const pcl::PointCloud<PointT> &cloud, 
                            const std::vector<int> &indices,
                            const int precision)

io/include/pcl/io/impl/point_cloud_image_extractors.hpp - 
pcl::io::PointCloudImageExtractor<PointT>::extract (const PointCloud& cloud, pcl::PCLImage& img)

visualization/include/pcl/visualization/impl/image_viewer.hpp - 
pcl::visualization::ImageViewer::addRGBImage (const pcl::PointCloud<T> &cloud,
                                              const std::string &layer_id,
                                              double opacity)

visualization/src/image_viewer.cpp - 
pcl::visualization::ImageViewer::addMonoImage (
    const pcl::PointCloud<pcl::Intensity> &cloud,
    const std::string &layer_id, double opacity)
pcl::visualization::ImageViewer::addMonoImage (
    const pcl::PointCloud<pcl::Intensity8u> &cloud,
    const std::string &layer_id, double opacity)
    
 suggested process for modification;
      1. Add API for index_t
      2. Deprecate + SFINAE out existing function
      3. Later remove deprecation
*/

Reply:
@SergioRAgostinho

Reply:
Answered above. Don't spam the code, just keep track of these items in your trello project.